PluralKit.Bot/Commands/Member.cs

```diff
@@ -70,21 +77,115 @@ public Member(EmbedService embeds, IDatabase db, ModelRepository repo, HttpClien
+                    .Color(ctx.System.Color?.ToDiscordColor() ?? DiscordUtils.Gray)
+                    .Description(ctx.System.WelcomeMessage)
+                    .Footer(new($"System ID: {ctx.System.Hid}"));
+
+                switch (ctx.System.WelcomeMessageMode) {
+                    case WelcomeMessageMode.Off: break;
+                    case WelcomeMessageMode.Inline:
+                    {
+                        embed = eb.Build();
+                        break;
+                    }
+                    case WelcomeMessageMode.DM:
+                    {
+                        try {
+                            var dm = await ctx.Cache.GetOrCreateDmChannel(ctx.Rest, ctx.Author.Id);
+                            await ctx.Rest.CreateMessage(dm.Id, new MessageRequest{
+                                Embed = eb.Build()
+                            });
+                        }
+                        catch (ForbiddenException)
+                        {
+                            if (ctx.Channel.Type != Channel.ChannelType.Dm)
+                                text += $"\n{Emojis.Error} Could not send welcome message in DMs. Are your DMs closed?";
+                        }
+                        break;
+                    }
+                    case WelcomeMessageMode.CustomChannel:
+                    {
+                        // This shouldn't ever happen, but if it does, act as if there's no message set at all.
+                        if (ctx.System.WelcomeMessageChannel == null) break;
+
+                        try {
+                            var channel = await ctx.Cache.GetOrFetchChannel(ctx.Rest, ctx.System.WelcomeMessageChannel.Value);
+                            if (channel == null)
+                            {
+                                text += $"\n{Emojis.Error} Could not find your welcome channel, has it been deleted?";
+                                break;
+                            }
+
+                            // GuildId should never be null
+                            if (channel?.GuildId == null) {
+                                text += $"\n{Emojis.Error} Your welcome channel isn't in a server.";
+                                break;
+                            }
+
+                            var guildMember = await ctx.Rest.GetGuildMember(channel?.GuildId ?? 0, ctx.Author.Id);
+                            if (guildMember == null) {
+                                text += $"\n{Emojis.Error} You're not in the server your welcome channel is in.";
+                                break;
+                            }
+
+                            var perms = ctx.Cache.PermissionsFor(channel.Id, guildMember);
+
+                            if (!perms.HasFlag(PermissionSet.ViewChannel | PermissionSet.ReadMessageHistory))
+                            {
+                                text += $"\n{Emojis.Error} You can't read messages in your welcome channel.";
```

Comment:
Please copy the error message from SystemEdit as to not expose info that may not be public.
```suggestion
                                text += $"\n{Emojis.Error} Could not find welcome channel, you are not in the server the welcome channel is in or you do not have permissions to view the welcome channel..";
```